src/Agent.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -65,9 +67,75 @@ public AgentSettings LoadSettings()
             return settings;
         }
 
+        private void checkAgentRootDirectorySecure()
+        {
+            Trace.Info(nameof(checkAgentRootDirectorySecure));
+
+            try
+            {
+                string rootDirPath = HostContext.GetDirectory(WellKnownDirectory.Root);
+
+                if (!String.IsNullOrEmpty(rootDirPath))
+                {
+                    // Get info about root folder
+                    DirectoryInfo dirInfo = new DirectoryInfo(rootDirPath);
+
+                    // Get directory access control list 
+                    DirectorySecurity directorySecurityInfo = dirInfo.GetAccessControl();
+                    AuthorizationRuleCollection dirAccessRules = directorySecurityInfo.GetAccessRules(true, true, typeof(NTAccount));
+
+
+                    // Get identity reference of the BUILTIN\Users group
+                    IdentityReference bulitInUsersGroup = new SecurityIdentifier(WellKnownSidType.BuiltinUsersSid, null).Translate(typeof(NTAccount));
+
+                    List<FileSystemAccessRule> potentiallyInsecureRules = new List<FileSystemAccessRule>();
+
+                    // Check if BUILTIN\Users group have modify/write rights for the agent root folder
+                    foreach (FileSystemAccessRule accessRule in dirAccessRules)
```

Comment:
Can we use PLINQ here - instead of searching in loop?

Reply:
Thanks for the suggestion! I replaced `foreach` with `PLINQ`.